scripts/lint_filename.go

```diff
@@ -0,0 +1,43 @@
+/*
+SPDX-FileCopyrightText: 2023 The Pion community <https://pion.ly>
+SPDX-License-Identifier: MIT
+*/
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+)
+
+func main() {
+	parentDir := filepath.Dir(os.Args[0])
+	goRegex := regexp.MustCompile(`^[a-zA-Z][a-zA-Z0-9_]*\.go$`)
+
+	err := filepath.Walk(parentDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		if filepath.Ext(path) != ".go" {
+			return nil
+		}
+
+		filename := filepath.Base(path)
+
+		if !goRegex.MatchString(filename) {
+			return fmt.Errorf("%s is not a valid filename for Go code, only alpha, numbers and underscores are supported", filename)
+		}
+
+		return nil
+	})
+	if err != nil {
```

Comment:
```suggestion
	}); err != nil {
```

I usually prefer this shorter form. But its a personal preference. Feel free to accept or decline this suggestion how you like :)